packages/catalog-model/src/kinds/util.ts

```diff
@@ -22,9 +22,18 @@ import { KindValidator } from './types';
 // exported kind validators have the `KindValidator` signature which is
 // different. So let's postpone that change until a later time.
 export function ajvCompiledJsonSchemaValidator(schema: unknown): KindValidator {
-  const validator = entityKindSchemaValidator(schema);
+  let cachedValidator = null;
+  const memoizedValidator = () => {
```

Comment:
No need for this extra function really, could just inline the contents of it within `check` right?

Reply:
Had a brain hiccup I think, thanks for calling this out.